libs/crucible_proc_macros/src/lib.rs

```diff
@@ -321,3 +334,89 @@ pub fn crux_spec_for(args: TokenStream, input: TokenStream) -> TokenStream {
     eprintln!("output = {}", tokens);
     tokens.into()
 }
+
+/// Adds support for `#[cfg_attr(crux, derive(Symbolic))]`
+/// 
+/// This generates an implementation of the Symbolic trait that
+/// is completely symbolic. All fields will be generated using
+/// their symbolic instances. In the case of enumerations, all
+/// variants can be returned. Unions are not supported.
+#[proc_macro_derive(Symbolic)]
+pub fn symbolic_derive(input: TokenStream) -> TokenStream {
+    let input = parse_macro_input!(input as DeriveInput);
+    let name = input.ident;
+
+    let body = match input.data {
+        Data::Struct(data_struct) => {
+            match &data_struct.fields {
+                Fields::Named(fields_named) => {
+                    let field_inits = fields_named.named.iter().map(|f| {
+                        let fname = &f.ident;
+                        quote! { #fname: crucible::Symbolic::symbolic(desc) }
+                    });
+                    quote! { Self { #( #field_inits, )* } }
+                }
+                Fields::Unnamed(fields_unnamed) => {
+                    let field_inits = fields_unnamed.unnamed.iter().map(|_| {
+                        quote! { crucible::Symbolic::symbolic(desc) }
+                    });
+                    quote! { Self( #( #field_inits, )* ) }
+                }
+                Fields::Unit => {
+                    quote! { Self }
+                }
+            }
+        }
+
+        Data::Enum(data_enum) => {
+            let variants: Vec<_> = data_enum.variants.iter().collect();
+            let arms = variants.iter().enumerate().map(|(i, v)| {
+                let vname = &v.ident;
+                match &v.fields {
+                    Fields::Named(fields_named) => {
+                        let field_inits = fields_named.named.iter().map(|f| {
+                            let fname = &f.ident;
+                            quote! { #fname: crucible::Symbolic::symbolic(desc) }
+                        });
+                        quote! { #i => Self::#vname { #( #field_inits, )* } }
+                    }
+                    Fields::Unnamed(fields_unnamed) => {
+                        let field_inits = fields_unnamed.unnamed.iter().map(|_| {
+                            quote! { crucible::Symbolic::symbolic(desc) }
+                        });
+                        quote! { #i => Self::#vname( #( #field_inits, )* ) }
+                    }
+                    Fields::Unit => {
+                        quote! { #i => Self::#vname }
+                    }
+                }
+            });
+
+            let n = variants.len();
+            quote! {
+                {
+                    let variant: usize = crucible::Symbolic::symbolic_where("variant", |&i| i < #n);
+                    match variant {
+                        #( #arms, )*
+                        _ => crucible::crucible_assume_unreachable!(),
+                    }
```

Comment:
Currently the `i < n` constraint is being assumed in two places, once in `symbolic_where` and again through the use of `crucible_assume_unreachable!`.  Can you remove one or the other, so they can't get out of sync?  For the second one, I think there's a `crucible_assert_unreachable!` you could replace it with.